scripts/reana_bench.py

```diff
@@ -218,136 +221,248 @@ def _clean_results(df: pd.DataFrame) -> pd.DataFrame:
 
-    df["runtime"] = df["ended"] - df["started"]
+    df["runtime"] = _convert_str_date_to_epoch(
+        df["ended"]
+    ) - _convert_str_date_to_epoch(df["started"])
     df["runtime"] = df["runtime"].astype(int)
     return df
 
 
-def _max_min_mean_median(series: pd.Series) -> (int, int, int, int):
-    max_value = series.max()
-    min_value = series.min()
-    mean_value = series.mean()
-    median_value = series.median()
-    return int(max_value), int(min_value), int(mean_value), int(median_value)
+def _build_execution_progress_plot(
+    df: pd.DataFrame, plot_parameters: Dict
+) -> (str, Figure):
+    title = plot_parameters["title"]
+    interval = plot_parameters["time_interval"]
+
+    fig, ax = plt.subplots(figsize=(8, 4), dpi=200, constrained_layout=True)
+
+    for index, row in df.iterrows():
+        created_date = datetime.strptime(row["created"], DATETIME_FORMAT)
+        started_date = datetime.strptime(row["started"], DATETIME_FORMAT)
+        ended_date = datetime.strptime(row["ended"], DATETIME_FORMAT)
+        start_submit_date = datetime.strptime(row["submit_date"], DATETIME_FORMAT)
+        workflow_number = row["workflow_number"]
+
+        # add created point
+        ax.plot(
+            created_date,
+            workflow_number,
+            ".",
+            markerfacecolor="grey",
+            markersize=1,
+            color="grey",
+            label="1-created",
+        )
 
+        # add asked to start point
+        ax.plot(
+            start_submit_date,
+            workflow_number,
+            ".",
+            markerfacecolor="darkorange",
+            markersize=1,
+            color="darkorange",
+            label="2-asked to start",
+        )
 
-def _execution_progress_plot(path: Path, df: pd.DataFrame, title: str) -> None:
-    plt.clf()
+        # add pending line
+        ax.hlines(
+            workflow_number,
+            xmin=start_submit_date,
+            xmax=started_date,
+            colors=["orange"],
+            label="3-pending",
+        )
 
-    sorted_df = df.sort_values(["submit_date", "submit_number"])
+        # add started point
+        ax.plot(
+            started_date,
+            workflow_number,
+            ".",
+            markerfacecolor="darkblue",
+            markersize=1,
+            color="darkblue",
+            label="4-started",
+        )
 
-    submit_offset = sorted_df["submit_date"] - sorted_df["submit_date"].iloc[0]
+        # add running line
+        ax.hlines(
+            workflow_number,
+            xmin=started_date,
+            xmax=ended_date,
+            colors=["blue"],
+            label="5-running",
+        )
 
-    # dummy x axis
-    numbers = [int(i) for i in range(0, len(sorted_df))]
+        # add finished point
+        ax.plot(
+            ended_date,
+            workflow_number,
+            ".",
+            markerfacecolor="lightgreen",
+            markersize=1,
+            color="lightgreen",
```

Comment:
The dark blue reads a bit hard... What about lightblue instead of lightgreen to mark end? Or darkgreen instead of darkblue to mark the start?  (just cosmetics we can change later)

![pMSSM-NoSys-1200Mi-10nodeb-56nodej-250max-21k8s-20211104-0sleep-1k_execution_progress_250_500](https://user-images.githubusercontent.com/517546/140516741-d1e893e7-816f-4e02-b613-34751dfb1842.png)

Reply:
Honestly, I don't have any preferences. I can try the colors that you suggested.